### PR TITLE
Ut refactor

### PR DESCRIPTION
**Pull request by rungokarol**

### CODE REVIEW

File: server/controler/login.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+var minLength = 3
+
+func (env *Env) Login(responseWriter http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		// http.Error(responseWriter,
+		// http.StatusText(http.StatusMethodNotAllowed),
+		// http.StatusMethodNotAllowed)
+		responseWriter.WriteHeader(http.StatusMethodNotAllowed)
```
Comment (SirWojtek) [inline server/controler/login.go:27]: So the difference between the commented section and the current implementation is that `WriteHeader` fills only error code in the response header whilst the previous code was also set some common error message in the response body. A slightly different, I guess we can stay with the header-only version.

File: server/controler/login_test.go [hunk 2]
```diff
@@ -61,5 +64,14 @@ func (suite *LoginReqTestSuite) TestRejectWhenBodyIsNotJson() {
 	assert.Equal(suite.T(), http.StatusBadRequest, suite.rr.Code)
 }
 
+func (suite *LoginReqTestSuite) TestRejectIfUserNofFoundInDataStore() {
+	jsonBody, _ := json.Marshal(map[string]string{"username": "foo", "password": "bar"})
```
Comment (SirWojtek) [inline server/controler/login_test.go:68]: I think it'd be better to export `loginReq` type and use it here

Comment (rungokarol) [reply server/controler/login_test.go:68]: you're 100% right

File: server/controler/login.go (hunk 1 above)
Comment (rungokarol) [reply server/controler/login.go:27]: Will error message be visible in console during "curl" testing?

File: server/controler/login.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		return
+	} else if user == nil {
+		http.Error(responseWriter, "User not found", http.StatusNotFound) //not sure if correct status
+		return
+	}
+
+	if err := bcrypt.CompareHashAndPassword([]byte(user.PasswordHash),
+		[]byte(loginReq.Password)); err != nil {
+		http.Error(responseWriter,
+			"Wrong username or password",
```
Comment (rungokarol) [inline server/controler/login.go:53]: what to do with these cases when message and code differs?

File: server/controler/login.go (hunk 1 above)
Comment (SirWojtek) [reply server/controler/login.go:27]: Not sure, worth checking that. The general rule is to return HTTP error code + add a more detailed description in the body of the response if possible.

File: server/controler/login.go (hunk 3 above)
Comment (SirWojtek) [reply server/controler/login.go:53]: As I wrote above:
> The general rule is to return HTTP error code + add a more detailed description in the body of the response if possible.

I think it's a good practice adding a description to the error response.

**Pull request merged**